src/org/labkey/test/tests/DomainDesignerTest.java

```diff
@@ -1498,6 +1507,172 @@ public void addUpdateRemoveConditionalFormat() throws Exception
         assertEquals(true, editedAquamap.get("bold"));
     }
 
+
+    @Test
+    public void testExportFields() throws Exception
+    {
+        String listName = "exportFieldsTestList";
+
+        FieldDefinition.LookupInfo lookupInfo = new FieldDefinition.LookupInfo(getProjectName(), "lists", listName);
+        ConditionalFormatFilter eqFive = new ConditionalFormatFilter("5", Filter.Operator.EQUAL);
+        FieldDefinition.RangeValidator soonValidator = new FieldDefinition.RangeValidator("soonValidator", "between now and 2 days from now", "aaaa",
+                FieldDefinition.RangeType.GTE, "2020-11-04",
+                FieldDefinition.RangeType.LTE, "2020-11-06");
+        TestDataGenerator dgen = new TestDataGenerator(lookupInfo)
+                .withColumns(List.of(
+                        new FieldDefinition("textField", FieldDefinition.ColumnType.String)
+                            .setDescription("is texty").setLabel("text field"),
+                        new FieldDefinition("booleanField", FieldDefinition.ColumnType.Boolean)
+                            .setLabel("Boolean Field")
+                            .setRequired(true)
+                            .setURL("fake/list-def.url"),
+                        new FieldDefinition("attachmentField", FieldDefinition.ColumnType.Attachment)
+                            .setHidden(true),
+                        new FieldDefinition("decimalField", FieldDefinition.ColumnType.Decimal).
+                            setScale(1234).setFormat("0.####"),
+                        new FieldDefinition("intField", FieldDefinition.ColumnType.Integer)
+                            .setMvEnabled(true)
+                            .setConditionalFormats(List.of(new ConditionalFormat(List.of(eqFive), true, true, false))),
+                        new FieldDefinition("dateTimeField", FieldDefinition.ColumnType.DateAndTime)
+                            .setValidators(List.of(soonValidator))));
+        dgen.createList(createDefaultConnection(), "Key");
```

Comment:
The one "extra" thing for lists is that the exported field definitions JSON has an `isPrimaryKey` property set to true for the key field. Can we be sure to check that in this list definition export case?

Reply:
thanks, I'll add checks for this.
Unfortunately, that's not a field on the PropertyDescriptor class (tho really, it should be)